### PR TITLE
OCPCRT-193: add support for rosa

### PR DESCRIPTION
**Pull request by AlexNPavel**
This PR adds a `rosa create <version>` command that allows a user whose ID is in a whitelist to launch clusters in the ROSA environment. The `auth` and `done` commands also work on ROSA clusters and if a user has a ROSA cluster, the cluster will show up in the `list` command. The chat-bot keeps passwords that it creates in a secret. When a cluster ID in the secret no longer appears in the list of clusters returned by ROSA, commands to clean up created account-roles and oidc-provider are run.

The default timeout for rosa clusters for now will be 6 hours.

### CODE REVIEW

File: hack/run.sh [hunk 1]
```diff
@@ -21,8 +23,12 @@ make
...
   --prow-config ${work_dir}/../release/core-services/prow/02_config/_config.yaml \
   --workflow-config-path ${work_dir}/../release/core-services/ci-chat-bot/workflows-config.yaml \
+  --rosa-user-allowlist-path ${work_dir}/../release/core-services/ci-chat-bot/rosa-user-allowlist.yaml \
+  --rosa-subnetlist-path $tmp_subnets \
   --lease-server-credentials-file $tmp_boskos \
   --override-launch-label "ci-chat-bot-${USER}.openshift.io/launch" \
+  --override-rosa-secret-name "ci-chat-bot-rosa-clusters-${USER}" \
   --kubeconfig-dir $tmp_kube \
   --kubeconfig-suffix=.config \
+  --impersonate-admin \
```
Comment (bradmwilliams) [inline hack/run.sh:33]: AFAICT, This flag is no longer supported.

**Review by bradmwilliams — CHANGES_REQUESTED** [review]
Just one little nit.  Other than that it looks great!

**Comment by bradmwilliams** [comment]
There was another test-infra bump this morning, so you'll probably have to rebase as well.  Hopefully, it won't cause you to much grief!

**Comment by AlexNPavel** [comment]
I rebased to the latest master (no issues from the test-infra bump) and removed the unused flag

File: cmd/ci-chat-bot/main.go [hunk 2]
```diff
@@ -103,6 +108,9 @@ func run() error {
 	pflag.StringVar(&opt.leaseServer, "lease-server", citools.URLForService(citools.ServiceBoskos), "Address of the server that manages leases. Used to identify accounts with more available leases.")
 	pflag.StringVar(&opt.leaseServerCredentialsFile, "lease-server-credentials-file", "", "The path to credentials file used to access the lease server. The content is of the form <username>:<password>.")
 	pflag.StringVar(&opt.overrideLaunchLabel, "override-launch-label", "", "Override the default launch label for jobs. Used for local debugging.")
+	pflag.StringVar(&opt.overrideRosaSecretName, "override-rosa-secret-name", "", "Override the default secert name for rosa cluster tracking. Used for local debugging.")
```
Comment (bradmwilliams) [inline cmd/ci-chat-bot/main.go:111]: Typo: `secret` is misspelled

**Review by bradmwilliams — CHANGES_REQUESTED** [review]
Just a typo, but other than that, looks good!

**Pull request merged**